components/intercom/new-admin-reply.js

```diff
@@ -0,0 +1,74 @@
+const intercom = require("https://github.com/PipedreamHQ/pipedream/components/intercom/intercom.app.js");
+
+module.exports = {
+  name: "New Reply From Admin",
+  description: "Emits an event each time an admin replies to a conversation.",
+  version: "0.0.1",
+  dedupe: "unique",
+  props: {
+    intercom,
+    db: "$.service.db",
+    timer: {
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 60 * 15,
+      },
+    },
+  },
+
+  async run(event) {
+    const monthAgo = this.intercom.monthAgo();
+    let lastAdminReplyAt =
+      this.db.get("lastAdminReplyAt") || Math.floor(monthAgo / 1000);
+    lastAdminReplyAt = Math.floor(monthAgo / 1000);
+    const data = {
+      query: {
+        field: "statistics.last_admin_reply_at",
+        operator: ">",
+        value: lastAdminReplyAt,
+      },
+    };
+
+    let results = null;
+    let starting_after = null;
+
+    while (
+      !results ||
+      (results.data.pages.next !== null &&
+        results.data.pages.next !== undefined)
+    ) {
+      if (results) starting_after = results.data.pages.next.starting_after;
+      results = await this.intercom.searchConversations(data, starting_after);
+
+      for (const conversation of results.data.conversations) {
+        if (conversation.statistics.last_admin_reply_at > lastAdminReplyAt)
+          lastAdminReplyAt = conversation.statistics.last_admin_reply_at;
+        let conversationData = (
+          await this.intercom.getConversation(conversation.id)
+        ).data;
+        let total_count = conversationData.conversation_parts.total_count;
+        if (
+          total_count > 0 &&
+          conversationData.conversation_parts.conversation_parts[
+            total_count - 1
+          ].body != null
```

Comment:
This might be a good use case for Lodash's `get` method, since in theory `conversationData.conversation_parts.conversation_parts[total_count - 1]` might not exist and you might get an error at this step. I'd suggest doing something like:

```
const get = require("lodash.get")
const conversationBody = get(conversationData, "conversation_parts.conversation_parts[total_count - 1].body")
if (total_count > 0 && conversationBody) { ...
```